lib/dry/result_matcher/evaluator.rb

```diff
@@ -0,0 +1,38 @@
+module Dry
+  module ResultMatcher
+    class Evaluator < BasicObject
+      def initialize(result, cases)
+        @cases = cases
+        @result = result
+        @matched = false
+        @output = nil
+      end
+
+      def call(&block)
+        block.call(self)
```

Comment:
@timriley you can just `yield(self)` here. On the one hand using `&block` is more consistent but on the other hand it forces ruby to wrap a block with a Proc object that leads to unnecessary heap allocation. See for details https://github.com/JuanitoFatas/fast-ruby#proccall-and-block-arguments-vs-yieldcode